src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/service/WitnessesService.java

```diff
@@ -0,0 +1,40 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.service;
+
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.ObjectUtils.defaultIfNull;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.WITNESS_DETAILS;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.WitnessDetails;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.IdValue;
+
+public class WitnessesService {
```

Comment:
can we use lombok annotations here? NoArgsConstructor

Reply:
This class has a private no args constructor to avoid possibility of instantiation. Using Lombok @NoArgsConstructor will generate a public constructor.